src/components/form-helpers/CloudTagField/index.test.js

```diff
@@ -0,0 +1,234 @@
+
+    // Arrange
+    const emailField = screen.getByRole('textbox');
+    const addTagButton = screen.getByRole('button', { name: 'add tag' });
+
+    // add first tag (simulated with click event)
+    expect(addTagButton).toHaveAttribute('disabled');
+    await fireEvent.change(emailField, { target: { value: email1 } });
+    expect(emailField.value).toBe(email1);
+    expect(addTagButton).not.toHaveAttribute('disabled');
+    await fireEvent.click(addTagButton);
+
+    // add second tag (simulated with enter event)
+    expect(addTagButton).toHaveAttribute('disabled');
+    await fireEvent.change(emailField, { target: { value: email2 } });
+    expect(emailField.value).toBe(email2);
+    expect(addTagButton).not.toHaveAttribute('disabled');
+    await fireEvent.keyDown(emailField, { key: 'Enter', code: 'Enter' });
+
+    const cloudTags = screen.getByRole('list', { name: 'cloud tags' });
+    expect(cloudTags).toBeInTheDocument();
+    matchTags(cloudTags, [email1, email2]);
+
+    expect(onSubmit).toHaveBeenCalledTimes(0);
+    const submitButton = screen.getByRole('button', { name: /Save/i });
+    await act(async () => fireEvent.click(submitButton));
+    expect(onSubmit).toHaveBeenCalledTimes(1);
+
+    expect(onSubmit.mock.calls[0][0]).toEqual({
+      [fieldName]: [email1, email2],
+    });
+  });
+
+  it('should shown errors message when tag to add is not an email', async () => {
+    // Arrange
+    const initialValues = [];
+    const emailInvalid = 'fake-text';
+    const emailValid = 'emailz@gmail.com';
+    const onSubmit = jest.fn();
+    const formikConfig = {
+      initialValues: { [fieldName]: initialValues },
+      onSubmit,
+    };
+
+    // Act
+    render(<WrapperComponent formikConfig={formikConfig} validateTag={validateEmail} />);
+
+    // Arrange
+    const emailField = screen.getByRole('textbox');
+    const addTagButton = screen.getByRole('button', { name: 'add tag' });
+
+    // add tag (simulated with click event)
+    expect(addTagButton).toHaveAttribute('disabled');
+    await fireEvent.change(emailField, { target: { value: emailInvalid } });
+    expect(emailField.value).toBe(emailInvalid);
+    expect(addTagButton).not.toHaveAttribute('disabled');
+    await fireEvent.click(addTagButton);
+
+    const cloudTags = screen.queryByRole('list', { name: 'cloud tags' });
+    const errors = screen.queryByRole('alert');
+
+    expect(errors).toBeInTheDocument();
+    expect(cloudTags).not.toBeInTheDocument();
+
+    // add tag (simulated with enter event)
+    await fireEvent.change(emailField, { target: { value: emailValid } });
+    expect(emailField.value).toBe(emailValid);
+    expect(addTagButton).not.toHaveAttribute('disabled');
+
+    fireEvent.keyDown(emailField, { key: 'Enter', code: 'Enter' });
+
+    expect(errors).not.toBeInTheDocument();
+    expect(await screen.findByRole('list', { name: 'cloud tags' })).toBeInTheDocument();
+  });
```

Comment:
Here you could add some tests for contact policy like cases, with a validateFunction that checks if the text is within an array list, and instead of having an input to add emails we have a button that adds elements in an array. Is it possible?